src/analysis/retail/shaman/restoration/modules/dragonflight/Season4Tier.tsx

```diff
@@ -0,0 +1,176 @@
+import { TIERS } from 'game/TIERS';
+import Analyzer, { Options, SELECTED_PLAYER } from 'parser/core/Analyzer';
+import STATISTIC_CATEGORY from 'parser/ui/STATISTIC_CATEGORY';
+import STATISTIC_ORDER from 'parser/ui/STATISTIC_ORDER';
+import Statistic from 'parser/ui/Statistic';
+import Abilities from 'parser/core/modules/Abilities';
+import { TALENTS_SHAMAN } from 'common/TALENTS';
+import Events, { HealEvent } from 'parser/core/Events';
+import HIT_TYPES from 'game/HIT_TYPES';
+import SpellUsable from 'parser/shared/modules/SpellUsable';
+import ItemPercentHealingDone from 'parser/ui/ItemPercentHealingDone';
+import { SHAMAN_DF4_ID } from 'common/ITEMS/dragonflight';
+import { formatNumber, formatPercentage } from 'common/format';
+import CritEffectBonus from 'parser/shared/modules/helpers/CritEffectBonus';
+import StatTracker from 'parser/shared/modules/StatTracker';
+import BoringItemSetValueText from 'parser/ui/BoringItemSetValueText';
+import { calculateEffectiveHealingFromCritIncrease } from 'parser/core/EventCalculateLib';
+
+/**
+ * **Resto Shaman Season 4 **
+ *
+ * 2pc: While [Cloudburst Totem / Healing Stream Totem] is active, your chance to critically strike is increased by 15%.
+ *
+ * 4pc : Your critical heals have 220% effectiveness instead of the usual 200%.
+ */
+enum ActiveTotem {
+  Cloudburst = 'cloudburst',
+  HealingStream = 'healing_stream',
+  None = 'none',
+}
+
+const CRIT_RATE_TOTEM_HEALING_INCREASE = 0.15;
+const CRIT_EFFECT_INCREASE_4PC = 0.2;
+
+class Season4Tier extends Analyzer {
+  static dependencies = {
+    abilities: Abilities,
+    spellUsable: SpellUsable,
+    statTracker: StatTracker,
+    critEffectBonus: CritEffectBonus,
+  };
+  protected abilities!: Abilities;
+  protected spellUsable!: SpellUsable;
+  protected statTracker!: StatTracker;
+  protected critEffectBonus!: CritEffectBonus;
+  protected criticalHealCount: number = 0;
+  protected increased2pcHealing: number = 0;
+  protected totalHealing: number = 0;
+  protected totalCritsFromBonus: number = 0;
+  protected activeTotem: ActiveTotem = ActiveTotem.None;
+  protected increased4pcHealing: number = 0;
+  protected has4pc: boolean;
+  constructor(options: Options) {
+    super(options);
+
+    this.active = this.selectedCombatant.has2PieceByTier(TIERS.DF4);
+    this.has4pc = this.selectedCombatant.has4PieceByTier(TIERS.DF4);
+
+    console.log(this.active);
+    console.log(this.has4pc);
+
+    this.addEventListener(Events.heal.by(SELECTED_PLAYER), this.onHeal);
+    this.addEventListener(
+      Events.applybuff.by(SELECTED_PLAYER).spell(TALENTS_SHAMAN.CLOUDBURST_TOTEM_TALENT),
+      this.onCloudburstApply,
+    );
+    this.addEventListener(
+      Events.removebuff.by(SELECTED_PLAYER).spell(TALENTS_SHAMAN.CLOUDBURST_TOTEM_TALENT),
+      this.onCloudburstRemove,
+    );
+
+    this.addEventListener(
+      Events.summon.by(SELECTED_PLAYER).spell(TALENTS_SHAMAN.HEALING_STREAM_TOTEM_SHARED_TALENT),
```

Comment:
@emallson , I'm having a hard time tracking when a totem is active. I have tried both `Events.applybuff` and `Events.summon` and neither work. Not sure if the issue is with detecting the talent or something else. I can see that in the warcraft logs the events I want is when the totem is summoned and when it dies. Any tips on how do I detect those events here?

<img width="792" alt="image" src="https://github.com/WoWAnalyzer/WoWAnalyzer/assets/6465613/76d38882-84cc-4483-93f8-bf8d63c878a4">



<img width="792" alt="image" src="https://github.com/WoWAnalyzer/WoWAnalyzer/assets/6465613/ff9d72d7-72b7-4543-9aca-28fb8131e4fe">

Reply:
can you link the log? odds are that the spell id is not the same as the spell listed in the talent. often for combining / charge granting talents the id is different

Reply:
Here is the log for the event. The `guid` is different both the spell and talent entries of "Healing Stream Totem". I could only find a reference to this guid in the classic components. Right now I'm checking against the hability name and it seems to work.

```json
ability: {name: 'Healing Stream Totem', guid: 5394, type: 8, abilityIcon: 'inv_spear_04.jpg'},
fight: 1,
sourceID: 1,
sourceIsFriendly: true,
targetID: 41,
targetInstance: 1,
targetIsFriendly: true,
timestamp: 647009,
type: "summon"
}

Reply:
this is a case where you need to add the spell definition to `common/SPELLS/shaman` with the correct id/name/icon

Reply:
often, spells have multiple effects and we put any that aren't in the auto-generated data under `SPELLS`

Reply:
Got it. I will need to get some logs with cloudburst as well to check the correct spell id. I will update it soon